app/controllers/comments_controller.rb

```diff
@@ -6,9 +6,9 @@ def create
     result = comment_creator.call
 
     if result.success
-      redirect_to submission_path(submission_id), notice: 'Comment was successfully created.'
+      redirect_to request.referer, notice: 'Comment was successfully created.'
```

Comment:
It is not so important in our case, but I will comment it here, for future:
`request.referer` is the data taken from a user, it can be potentially mangled by the attacker.
Ideally, you should validate if the referer uses the same host as you app before redirecting.